### PR TITLE
feat:Core Crypto proteus initiation

### PR DESCRIPTION
**Pull request by tlebon**
## Pull Request Checklist

- [ ] My code is covered by tests
- [x] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/core/src/Account.ts [hunk 1]
```diff
@@ -245,43 +250,67 @@ export class Account<T = any> extends EventEmitter {
...
-    // Assumption: client gets only initialized once
-    if (initClient) {
-      await this.initClient({clientType});
+      // client gets initialized twice: once for proteus/ once for mls
+      this.proteusClient = await CoreCrypto.init({
+        databaseName: `corecrypto-${this.generateDbName(context)}-proteus`,
+        key: Encoder.toBase64(key).asString,
+        clientId: `PROTEUS_${context.clientId}`,
+        wasmFilePath: this.cryptoProtocolConfig!.coreCrypoWasmFilePath,
+        ...(entropyData && {entropySeed: entropyData}),
```
Comment (tlebon) [inline packages/core/src/Account.ts:278]: not really sure what to do about this right now.

Comment (atomrc) [reply packages/core/src/Account.ts:278]: I think it should only be in the `register` case, doesn't it? 
Once the identification of the device has been created, we don't need extra entropy anymore. 

When we call `init` we know that the device has been created, thus we do not need extra entropy

File: packages/core/src/Account.ts [hunk 2]
```diff
@@ -245,43 +250,67 @@ export class Account<T = any> extends EventEmitter {
...
+        : await createEncryptedStore(dbName);
 
-        await this.apiClient.api.client.putClient(context.clientId!, {prekeys});
-        this.logger.debug(`Successfully uploaded '${prekeys.length}' PreKeys.`);
-      },
+      let key = await secretStore.getsecretValue(coreCryptoKeyId);
 
-      onNewSession: onNewClient,
-    });
+      if (!key) {
```
Comment (aweiss-dev) [inline packages/core/src/Account.ts:267]: Why do we call the same function twice if it fails to return a key to us? This code is hard to understand for me.

File: packages/core/src/Account.ts [hunk 3]
```diff
@@ -368,7 +397,11 @@ export class Account<T = any> extends EventEmitter {
         }
 
         this.logger.log('Last client was permanent - Deleting cryptography stores');
-        await this.service!.cryptography.deleteCryptographyStores();
+        if (this.cryptoProtocolConfig?.proteus) {
+          await this.logout(true);
+        } else {
+          await this.service!.cryptography.deleteCryptographyStores();
+        }
```
Comment (atomrc) [inline packages/core/src/Account.ts:404]: can't we always use `logout(true)` here and remove the condition? I guess `logout(true)` will take care of deleting the DB

File: packages/core/src/mls/MLSService/MLSService.ts [hunk 4]
```diff
@@ -92,17 +92,21 @@ export class MLSService {
...
-      clientIsExistingGroupUser: (client, otherClients) => {
-        const decoder = new TextDecoder();
-        const {user} = parseFullQualifiedClientId(decoder.decode(client));
-        return otherClients.some(client => {
-          const {user: otherUser} = parseFullQualifiedClientId(decoder.decode(client));
-          return otherUser.toLowerCase() === user.toLowerCase();
-        });
-      },
+      userAuthorize: (groupId, client, otherClients) =>
+        coreCryptoCallbacks.authorize(groupId, client) && this.clientIsExistingGroupUser(client, otherClients),
```
Comment (atomrc) [inline packages/core/src/mls/MLSService/MLSService.ts:108]: Humm I think this will need to be implemented on the webapp side, please revert this change

File: packages/core/src/Account.ts [hunk 5]
```diff
@@ -451,9 +480,11 @@ export class Account<T = any> extends EventEmitter {
     const loadedClient = await this.service!.client.getLocalClient();
     await this.apiClient.api.client.getClient(loadedClient.id);
     this.apiClient.context!.clientId = loadedClient.id;
-    await this.service!.cryptography.initCryptobox();
+    if (!this.cryptoProtocolConfig?.proteus && !this.cryptoProtocolConfig?.mls) {
+      await this.service!.cryptography.initCryptobox();
+    }
     if (this.cryptoProtocolConfig?.mls && this.backendFeatures.supportsMLS) {
-      this.coreCryptoClient = await this.createMLSClient(
+      this.mlsClient = await this.createMLSClient(
```
[the remark this answers is not in the document]
Comment (aweiss-dev) [reply packages/core/src/Account.ts:487]: I agree, I would go a step further and rename createMLSClient() function to getCoreCryptoClientInstance()

**Comment by atomrc** [comment]
closed in favor of #4574
